File: src/test/java/net/xaviersala/Vaca/Model/VacaTests.java
package net.xaviersala.Vaca.Model;

import static org.junit.jupiter.api.Assertions.*;

import java.util.stream.Stream;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;
import org.mockito.Mockito;

class VacaTests {

	@BeforeEach
	void setUp() throws Exception {
	}

	static Stream<Arguments> CreatePesWithLitres() {
		return Stream.of(Arguments.of(12.0, 5.5), Arguments.of(5.0, 4.0));
	}

	@ParameterizedTest
	@MethodSource("CreatePesWithLitres")
	void testQueRetornaCorrectamentLaLlet(double pes, double litres) {

		// ARRANGE
		Raca racaFalsa = Mockito.mock(Raca.class);
		Mockito.when(racaFalsa.getLitresPerKg()).thenReturn(litres);

		Vaca sut = new Vaca("nom", pes, racaFalsa);

		// ACT
		double resultat = sut.GetLitres();

		// ASSERT
		assertEquals(pes * litres, resultat);
	}

}

File: Vaques.Tests/Model/VacaTests.cs
using System;
using Moq;
using Xunit;

namespace Vaques.Model
{
    public class VacaTests
    {
        [Theory]
        [InlineData(23, 12)]
        [InlineData(10, 10)]
        [InlineData(1, 12)]
        public void TestQueCalculaBeElsLitres(double pes, double litres)
        {
            // ARRANGE: Crear Raca. Els paràmetres no importen perquè els falsifico
            var raca = new Mock<Raca>("No importa", 0);
            raca.Setup(r => r.LitresPerKg).Returns(litres);

            var sut = new Vaca("X", pes, raca.Object);

            // ACT
            var resultat = sut.GetLitres();

            // ASSERT
            Assert.Equal(pes * litres, resultat);

        }

    }
}
